ee/clickhouse/queries/session_recordings/clickhouse_session_recording_list.py

```diff
@@ -1,119 +1,234 @@
+            any(distinct_id) as distinct_id,
+            COUNT((JSONExtractInt(snapshot_data, 'type') = 2 OR JSONExtractBool(snapshot_data, 'has_full_snapshot')) ? 1 : NULL) as full_snapshots
+        FROM session_recording_events
         WHERE
             team_id = %(team_id)s
             {events_timestamp_clause}
-            {event_filter_event_where_clause}
+        GROUP BY session_id
+        HAVING full_snapshots > 0
+        {recording_start_time_clause}
+        {duration_clause} 
+    ) AS session_recordings
+    ON session_recordings.distinct_id = filtered_events.distinct_id
+    JOIN (
+        SELECT * FROM person_distinct_id WHERE person_distinct_id.team_id = %(team_id)s
+    ) as person_distinct_id 
+    ON person_distinct_id.distinct_id = session_recordings.distinct_id
+    JOIN ({person_query}) as person ON person.id = person_distinct_id.person_id 
+    WHERE
+        empty(filtered_events.event) OR
+        (
+            filtered_events.timestamp >= session_recordings.start_time
+            AND filtered_events.timestamp <= session_recordings.end_time
+        )
+        {person_id_clause}
+    GROUP BY session_recordings.session_id
+    HAVING 1 = 1
+    {event_filter_aggregate_having_clause}
+    ORDER BY start_time DESC
+    LIMIT %(limit)s OFFSET %(offset)s
     """
 
-    def _get_distinct_id_clause(self) -> Tuple[Dict[str, Any], str]:
-        distinct_id_clause = ""
-        distinct_id_params = {}
-        if self._filter.person_uuid:
-            distinct_id_clause = f"AND distinct_id IN (SELECT distinct_id from person_distinct_id WHERE person_id = %(person_uuid)s AND team_id = %(team_id)s)"
-            distinct_id_params = {"person_uuid": self._filter.person_uuid, "team_id": self._team.pk}
-        return distinct_id_params, distinct_id_clause
-
-    def _get_events_query_with_aggregate_clauses(self) -> EventsQueryWithAggregateClausesSQL:
-        event_filters = format_event_filters(self._filter)
-        events_timestamp_params, events_timestamp_clause = self._get_events_timestamp_clause()
-        event_query = self._event_query.format(
-            events_timestamp_clause=events_timestamp_clause,
-            event_filter_event_select_clause=event_filters.event_select_clause,
-            event_filter_event_where_clause=event_filters.event_where_clause,
-        )
+    def _get_person_query(self,) -> Tuple[str, Dict[str, Any]]:
+        return ClickhousePersonQuery(filter=self._filter, team_id=self._team.pk).get_query()
 
-        params: Dict[str, Any] = {"team_id": self._team.pk, **events_timestamp_params, **event_filters.params}
+    def _has_entity_filters(self):
+        return self._filter.entities and len(self._filter.entities) > 0
 
-        return EventsQueryWithAggregateClausesSQL(
-            event_query, params, event_filters.aggregate_select_clause, event_filters.aggregate_where_clause,
+    def _get_limit(self):
```

Comment:
Nit: Make this into a `@property` and you can kill some variable assignments below.